src/backend/marsha/core/serializers/playlist.py

```diff
@@ -26,15 +32,47 @@ class Meta:
             "users",
         ]
 
+    portable_to = serializers.SerializerMethodField(read_only=False)
+
+    def get_portable_to(self, obj):
+        """Getter for portable_to attribute instead of PlaylistSerializer to prevent recursion."""
+        return obj.portable_to.values("id", "title")
+
     def create(self, validated_data):
         """Create the Playlist object with the passed data."""
         return Playlist.objects.create(**validated_data)
 
+    def update(self, instance, validated_data):
+        """
+        Update the Playlist object with the passed data.
+
+        Instance id is removed from portable_to attribute.
+        """
+        while True:
+            try:
+                validated_data.get("portable_to").remove(str(instance.id))
+            except (AttributeError, ValueError):
+                break
+        return super().update(instance, validated_data)
+
+    def to_internal_value(self, data):
+        """Override to add portable_to playlists."""
+        value: OrderedDict = super().to_internal_value(data)
+        if "portable_to" in data.keys():
+            reachable_from_playlist_ids = data.get("portable_to")
+            for playlist_id in reachable_from_playlist_ids:
+                try:
+                    Playlist.objects.get(pk=playlist_id)
+                except Playlist.DoesNotExist:
+                    self.fail("playlist_not_exists")
```

Comment:
maybe this would be enough in one query and would also block duplicates:
```django
if Playlist.objects.filter(id__in=data.get("portable_to", []).count() != len(reachable_from_playlist_ids)
```
To make sure, we need tests:
- when portable_to is not set
- when portable_to has duplicates
- when portable to has unknown ids
- what else?

Reply:
Many tests (maybe too much ?) have been added to src/backend/marsha/core/tests/test_api_playlist_portability.py